### PR TITLE
Update doc with schema less mode

### LINKED ISSUE #32168: Documentation

### PR DESCRIPTION
**Pull request by rodireich**
Doc update

### CODE REVIEW

**Review by stephane-airbyte — APPROVED** [review]

File: docs/integrations/sources/mongodb-v2.md [hunk 1]
```diff
@@ -160,10 +160,28 @@ The MongoDB source utilizes change data capture (CDC) as a reliable way to keep
 
 Airbyte utilizes [the change streams feature](https://www.mongodb.com/docs/manual/changeStreams/) of a [MongoDB replica set](https://www.mongodb.com/docs/manual/replication/) to incrementally capture inserts, updates and deletes using a replication plugin. To learn more how Airbyte implements CDC, refer to [Change Data Capture (CDC)](https://docs.airbyte.com/understanding-airbyte/cdc/).
 
+### No Schema Mode
```
Comment (akashkulk) [inline docs/integrations/sources/mongodb-v2.md:163]: is this the final name we settled on? I remember schema-less/packed being thrown around

**Review by akashkulk — APPROVED** [review]

File: docs/integrations/sources/mongodb-v2.md (hunk 1 above)
Comment (rodireich) [reply docs/integrations/sources/mongodb-v2.md:163]: I don't think there's a term agreed upon.
"Wrapped mode" also suggested.

Title here is more of a description. not a name.
Does "Schema not enforced mode" sounds better?
"Schema Unenforced"?

Comment (Hesperide) [reply docs/integrations/sources/mongodb-v2.md:163]: What's the name in the product? We should match the wording customers see in the UI.

Comment (Hesperide) [inline docs/integrations/sources/mongodb-v2.md:163]: Can we change where this is laid out? The implication right now is customers need to pick between CDC and this new mode, which isn't quite true.

We should add a new section called # Schema Enforcement, or something like that, then lay out both options (strict schema + schemaless) in that section.

File: docs/integrations/sources/mongodb-v2.md [hunk 2]
```diff
@@ -160,10 +160,28 @@ The MongoDB source utilizes change data capture (CDC) as a reliable way to keep
...
+* Schema discovery uses [sampling](https://www.mongodb.com/docs/manual/reference/operator/aggregation/sample/) of the documents to collect all distinct top-level fields.  This value is universally applied to all collections discovered in the target database.  The approach is modelled after [MongoDB Compass sampling](https://www.mongodb.com/docs/compass/current/sampling/) and is used for efficiency.  By default, 10,000 documents are sampled.  This value can be increased up to 100,000 documents to increase the likelihood that all fields will be discovered.  However, the trade-off is time, as a higher value will take the process longer to sample the collection. 
+* When Running with Schema Enforced set to `false` there is no attempt to discover any schema. See more in [No Schema Mode](#No-Schema-Mode).
```
Comment (Hesperide) [inline docs/integrations/sources/mongodb-v2.md:184]: Should we reword this to say that you can't use column selection? If I understand correctly, that's the customer impact here.

File: docs/integrations/sources/mongodb-v2.md (hunk 1 above)
Comment (rodireich) [reply docs/integrations/sources/mongodb-v2.md:163]: In the product you have a new "Schema Enforced" field
image


This sector is about when it's not enforced.
I'm taking the other comment from before and create a "Schema enforcement" section

File: docs/integrations/sources/mongodb-v2.md (hunk 2 above)
Comment (rodireich) [reply docs/integrations/sources/mongodb-v2.md:184]: I'll add something on column selection.
I filed #32170 to deal with it (not an MVP for schemaless)

Comment (rodireich) [reply docs/integrations/sources/mongodb-v2.md]: I added something on lack of column selection in the schema enforecement section

I filed #32170 to deal with field de/selection in schemaless mode by the way

File: docs/integrations/sources/mongodb-v2.md (hunk 1 above)
Comment (rodireich) [reply docs/integrations/sources/mongodb-v2.md]: The new config field is "Schema Enforced"
Screenshot 2023-12-18 at 11 56 58 AM

**Comment by rodireich** [comment]
@Hesperide Please see if new layout and contents capture your notes. thx

**Comment by rodireich** [comment]
/connector-performance connector=connectors/source-postgres dataset=bottleneck_stream1

**Comment by rodireich** [comment]
/connector-performance connector=connectors/source-mongodb-v2
